### ISSUE TITLE
Utilizing default animations with headless tippy

### ISSUE THREAD

**Issue by ryanjmack** [atomiks/tippyjs-react#199]
Does headless tippy support the default animations?

I've tried utilizing the default animations with headless tippy to no avail.

I noticed that in the examples for headless animation [[1](https://codesandbox.io/s/vigilant-northcutt-7w3yr?file=/src/App.js)][[2](https://codesandbox.io/s/festive-fire-hcr47)] that `animation` takes a boolean prop instead of the animation name (e.g. `shift-towards-subtle`).

This is a contrived example to illustrate what I'm trying to do:

```
 <Tippy
        render={attrs => (
          <div className="tippy-box" data-animation="shift-towards-subtle" {...attrs}>
            <div className="tippy-content">
              {this.props.content}
            </div>
            <div data-popper-arrow="" className="tippy-arrow" />
          </div>
        )}
      >
        <span>I'm a tooltip!</span>
      </Tippy>
```

I've tried setting `data-animation` AND `animation` attributes on:
* `Tippy`
* inside of `render` 
* and on `.tippy-content` element

Thanks!

**Comment by atomiks** [comment]
I would use a spring library with Headless Tippy tbh, it's the biggest benefit of it. Especially since those libraries are useful in most other parts of the application so you'll be using them anyway.

You need to handle all the CSS animation logic yourself if you're using the default animations, which means setting `data-state="visible"` `onMount` and `data-state="hidden"` `onHide` on the `.tippy-box` element, and calling `unmount()` on CSS `transitionend`.

**Comment by ryanjmack** [comment]
Hi Atomiks,

I was just wondering if you ever had issues with the CSS translate property? I'm making a custom animation, very similar to the default `shift-towards-subtle` animation. 

On the very first trigger of any given tooltip, the opacity transitions in but not the translate. Every render afterwards behaves as expected.

Another contrived example:

```
class Tooltip extends PureComponent {

  constructor(props) {
    super(props);

    this.state = {
      isVisible: false,
    };

    this.onMount = this.onMount.bind(this);
    this.onHide = this.onHide.bind(this);
  }

  onMount() {
    this.setState({isVisible: true});
  }

  onHide(instance) {
    const unmountInstance = () => {
      instance.unmount();

      // need to control when we remove the listener because transitionend fires multiple times
      instance.popper.firstChild.removeEventListener('transitionend', unmountInstance);
    };

    instance.popper.firstChild.addEventListener('transitionend', unmountInstance);

    this.setState({isVisible: false});
  }

  render() {
    return (
       <Tippy  animation onMount={this.onMount} onHide={this.onHide}
        render={attrs => (
          <div className="tippy-box" data-state={this.state.isVisible ? 'visible' : 'hidden'} {...attrs}>
            <div className="tippy-content">
              {this.props.content}
            </div>
            <div data-popper-arrow="" className="tippy-arrow" />
          </div>
        )}>
        <span>I'm a tooltip!</span>
      </Tippy>
    )}
}
```

I tried debugging this extensively to no avail. I don't think there's an issue with the css since it works for every render, except the very first render.

here's a gif:
![out](https://user-images.githubusercontent.com/17010097/79825847-02273380-834f-11ea-99be-b28587dd472e.gif)

Thanks a bunch! I really appreciate your help! 😁

**Comment by atomiks** [comment]
Uhmm I remember having this problem many times in vanilla tippy but can't remember what caused it lol 😓 

Though why are you using `render` prop if you're just recreating Tippy's vanilla DOM element creation and using its animations? I would use Default Tippy. I'll try to debug this for you tho.

**Comment by atomiks** [comment]
Working version: https://codesandbox.io/s/sharp-wave-gj92x?file=/src/App.js

Looks like `transitionDuration: this.state.isVisible ? '400ms' : '0'` is the important part.

**Comment by ryanjmack** [comment]
Beautiful! It worked!

The reason we're going headless is because we need granular control over the class names that are applied to the elements that compose the tooltip.

We use CSS modules and rename our classes in our prod build pipeline. When we used the previous version of tippy-react, we had to modify our build tasks and use global css selectors for `tippy-*` classes. It sufficed, but with headless we can get rid of those workarounds and remove a dependency in our build pipeline!
